### PR TITLE
Add alias property in KeyStore options for net/http servers

### LINKED ISSUE #3926: Please enhance JksOptions to support selection of SSL key in keystore using an alias property
For context, I asked this question originally for Quarkus (see:  https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Any.20way.20to.20specify.20SSL.20key.20alias.20when.20using.20a.20keystore.20.3F ) and was asked to raise an issue for a Vert.x change.

There is currently no way to choose the SSL key alias for cases where a java key store may contain more than one key, thus no way to ensure the correct key is used when presenting the server certificate for SSL.

### PR DESCRIPTION
**Pull request by tsegismont**
Fixes #3926

This can be useful if users don't control the JKS file and it contains
multiple entries.

In this case, the SSL engine chooses the first one by default but it
might not be what the user expects.

### CODE REVIEW

File: src/main/java/io/vertx/core/net/KeyCertOptions.java
Comment (vietj) [inline src/main/java/io/vertx/core/net/KeyCertOptions.java:51]: is this used because the operation should only happen on the server ?

Comment (tsegismont) [reply src/main/java/io/vertx/core/net/KeyCertOptions.java:51]: Yes

**Comment by vietj** [comment]
I think we can avoid it since, the user simply should not set an alias on a
client JKS options

**Comment by tsegismont** [comment]
Done

>

**Comment by vietj** [comment]
shouldn't this option also apply to keystore in general ? e.g it should work too for `PfxOptions` or the generic `KeyStoreOptions` new in 4.0

**Comment by vietj** [comment]
@tsegismont can you make this option generic for any keystore ?

**Comment by tsegismont** [comment]
@vietj PTAL

**Comment by vietj** [comment]
I think we are missing in `ssl.txt` the steps you did to create the new keystore, so when it gets invalid, anyone knows the procedure to generate it.

**Comment by tsegismont** [comment]
@vietj I think I made them valid for the next 100 years ;-) But point taken.

**Comment by tsegismont** [comment]
@vietj done

**Comment by vietj** [comment]
waiting for CI green light and then we can merge this

**Review by vietj — APPROVED** [review]

**Pull request merged**
